control_plane/attachment_service/src/reconciler.rs

```diff
@@ -662,10 +682,26 @@ impl Reconciler {
     }
 }
 
+/// We tweak the externally-set TenantConfig while configuring
+/// locations, using our awareness of whether secondary locations
+/// are in use to automatically enable/disable heatmap uploads.
+fn ha_aware_config(config: &TenantConfig, has_secondaries: bool) -> TenantConfig {
+    let mut config = config.clone();
+    if has_secondaries {
+        if config.heatmap_period.is_none() {
+            config.heatmap_period = Some(DEFAULT_HEATMAP_PERIOD.to_string());
+        }
```

Comment:
>I guess we can't do it in the pageserver since we don't have knowledge of secondaries there.

Right, it's kind of a funny case, because the shards don't know about each other.  This could have just been a bool (and then let pageservers apply their own default Duration), but it's more flexible this way